api/http/management.go

```diff
@@ -27,3 +36,70 @@ func NewManagementAPI(app app.App) *ManagementAPI {
 		App: app,
 	}
 }
+
+func (api *ManagementAPI) SetConfiguration(c *gin.Context) {
+	var configuration model.Attributes
+
+	ctx := c.Request.Context()
+	devID, err := uuid.Parse(c.Param("device_id"))
+	if err != nil {
+		rest.RenderError(c,
+			http.StatusBadRequest,
+			errors.Wrap(err, "correctly formatted device id is needed"),
+		)
+		return
+	}
+
+	err = c.ShouldBindJSON(&configuration)
+	if err != nil {
+		rest.RenderError(c,
+			http.StatusBadRequest,
+			errors.Wrap(err, "malformed request body"),
+		)
+		return
+	}
+
+	for _, a := range configuration {
+		if err := a.Validate(); err != nil {
+			rest.RenderError(c,
+				http.StatusBadRequest,
+				errors.Wrap(err, "invalid request body"),
+			)
+		}
+	}
+
+	err = api.App.SetConfiguration(ctx, devID, configuration)
+	if err != nil {
+		c.Error(err) //nolint:errcheck
+		rest.RenderError(c,
+			http.StatusInternalServerError,
+			errors.New(http.StatusText(http.StatusInternalServerError)),
+		)
+		return
+	}
+	c.Status(http.StatusNoContent)
+}
+
+func (api *ManagementAPI) GetConfiguration(c *gin.Context) {
+	ctx := c.Request.Context()
+
+	devID, err := uuid.Parse(c.Param("device_id"))
+	if err != nil {
+		rest.RenderError(c,
+			http.StatusBadRequest,
+			errors.Wrap(err, "correctly formatted device id is needed"),
+		)
+		return
+	}
+
+	device, err := api.App.GetDevice(ctx, devID)
+	if err != nil {
+		rest.RenderError(c,
+			http.StatusNotFound,
+			errors.New(http.StatusText(http.StatusNotFound)),
+		)
+		return
```

Comment:
Sorry I didn't catch this in the first round, but we need to detect if the error is actually internal:
```suggestion
		switch cause := errors.Cause(err); cause {
		case store.ErrDeviceNoExist:
			rest.RenderError(c,
				http.StatusNotFound,
				errors.New(cause),
			)
		default:
			c.Error(err)
			rest.RenderError(c,
				http.StatusNotFound,
				errors.New(http.StatusText(http.StatusInternalServerError)),
			)
		}
		return
```

Reply:
fixed